fasta/assemblyStats.go

```diff
@@ -12,37 +12,48 @@ import (
 // AssemblyStats takes the path to a fasta file and a flag for whether lower case letters
 // should count as assembly gaps.  Five ints are returned, which encode:
 // the N50 size, half the size of the genome, size of the genome, size of the largest contig, and the number of contigs.
-func AssemblyStats(infile string, countLowerAsGaps bool) (int, int, int, int, int) {
+func AssemblyStats(infile string, countLowerAsGaps bool) (int, int, int, int, int, int) {
 	records := Read(infile)
-	var genomeLength int
 
 	contigList := MakeContigList(records, countLowerAsGaps)
 
-	for i := 0; i < len(contigList); i++ {
-		genomeLength += contigList[i]
-	}
+	genomeLength := calculateGenomeLength(contigList)
+
 	sort.Ints(contigList)
 	halfGenome := genomeLength / 2
 
-	N50 := CalculateN50(contigList, halfGenome)
+	N50, L50 := CalculateN50L50(contigList, halfGenome)
 	numContigs := len(contigList)
 	largestContig := contigList[len(contigList)-1]
 
-	return N50, halfGenome, genomeLength, largestContig, numContigs
+	return N50, L50, halfGenome, genomeLength, largestContig, numContigs
 }
 
-// CalculateN50 takes a slice of contig lengths and the size of half
-// the genome.  It returns the N50 size.
-func CalculateN50(contigList []int, halfGenome int) int {
+// calculateGenomeLength is a helper function that takes in a slice of int representing contigs in an assembly and returns the genome size (int)
+func calculateGenomeLength(contigList []int) int {
+	if len(contigList) == 0 {
+		log.Fatalf("Error: Cannot calculate genome length -- contig list is empty")
+	}
+	var genomeLength int
+	for i := 0; i < len(contigList); i++ {
+		genomeLength += contigList[i]
+	}
+	return genomeLength
+}
+
+// CalculateN50L50 takes a slice of contig lengths and the size of half the genome. It returns the N50 size and L50 size.
+func CalculateN50L50(contigList []int, halfGenome int) (int, int) {
```

Comment:
What do you think about splitting the N50 and L50 (s) into two functions?

Reply:
Yes its a good question, and I agree that it isn't very elegant. I opted to have them both come from the same function because the L50 is baked into the N50 function (essentially just counting how many times the loop was completed). If we broke them into two different functions we would sort of just be doing the same calculation and running the same code two times. Not sold on that though, happy to hear other opinions on this.

Reply:
I agree with the elegance point as well. I think for me it comes down to how frequently you're going to be using both of these statistics together. If you're going to want both of them most of the time anyway, then it feels slightly more efficient to keep them together. If you're expecting to want either N50 or L50 most of the time, then I'm in favor of splitting them up.

Reply:
We talked about this in code review today and decided that keeping them together probably makes sense unless there's a big efficiency loss involved in calculating the L50 along with the N50 (which seems unlikely)